### PR TITLE
tox and CI stuff

### PR DESCRIPTION
**Pull request by jerith**
We need to set up CI for vumi-go and tox is a good way to do that. This means a bunch of (mostly minor) changes and fixes to requirements, `setup.py`, etc.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
+                name = "%(name)s==%(version)s" % ver_match.groupdict()
+            install_requires.append(name)
+            dependency_links.append(requirement)
+        else:
+            install_requires.append(requirement)
+    return install_requires, dependency_links
+
+install_requires, dependency_links = parse_requirements("requirements.pip")
+print install_requires
+print dependency_links
```
Comment (smn) [inline setup.py]: stray prints

File: requirements.pip [hunk 2]
```diff
@@ -12,17 +12,19 @@ Fabric==1.2.2
...
--e git+git://github.com/praekelt/vxpolls.git@develop#egg=vxpolls
+https://github.com/praekelt/vumi/tarball/develop#egg=vumi-dev
+https://github.com/praekelt/vxpolls/tarball/develop#egg=vxpolls-dev
 Markdown==2.1.1
 django-registration==0.8
 # django_compressor==1.1.2 on Pypi has a bug, install from GitHub
 # until a new version has been published to pypi
--e git+git://github.com/jezdez/django_compressor@236e0d711#egg=compressor
+# -e git+git://github.com/jezdez/django_compressor@236e0d711#egg=compressor
+django_compressor>=1.2
```
Comment (smn) [inline requirements.pip:22]: This requirements can go but I suggest doing this in a separate PR as it also touches `go/settings.py`:

```
➜  go git:(feature/VUMIGO-176-endpoint-based-routing) ✗ git grep compress | grep py
go/base/widgets.py:            "%s/lib/codemirror-compressed.js" % (codemirror_path,),
go/settings.py:    'compressor.finders.CompressorFinder',
go/settings.py:    'compressor',
```

Comment (smn) [inline requirements.pip:22]: Created #227 to address this.

File: setup.py [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
+
+
+def parse_requirements(filename):
+    install_requires = []
+    dependency_links = []
+    for requirement in listify(filename):
+        if requirement.startswith("#"):
+            continue
+        if requirement.startswith("-e"):
+            continue
```
Comment (smn) [inline setup.py:19]: You've removed all the ones with `-e` should we log a warning here perhaps?

**Comment by smn** [comment]
:+1: in general, just a question about `-e` instead of skipping them silently.

File: setup.py (hunk 3 above)
Comment (jerith) [inline setup.py:19]: I don't know how to do that in a useful way without potentially breaking packaging.

**Pull request merged**